core/state_connector_test.go

```diff
@@ -0,0 +1,268 @@
+
+func (m *mockStateConnectorCaller) BlockContext() vm.BlockContext { return m.context }
+
+func buildStateConnectorMock(from string, addr common.Address) (*mockMessage, *mockStateConnectorCaller) {
+	msg := mockMessage{
+		from: common.BytesToAddress([]byte(from)),
+	}
+	scc := mockStateConnectorCaller{
+		msg: &msg,
+		context: vm.BlockContext{
+			Coinbase: addr,
+		},
+	}
+	return &msg, &scc
+}
+
+func TestStateTransition_FinalisePreviousRound(t *testing.T) {
+	currentRoundNumber := []byte("222")
+	testCoinbaseAddress := common.BigToAddress(big.NewInt(1000))
+	blockTime := big.NewInt(time.Date(2022, time.February, 28, 17, 1, 1, 0, time.UTC).Unix())
+
+	t.Run("coston chain", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("nominal case", func(t *testing.T) {
+			t.Parallel()
+
+			mockMsg, mockSCC := buildStateConnectorMock("coston chain", testCoinbaseAddress)
+
+			mockSCC.CallFunc = func(context vm.BlockContext, caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+				if caller.Address() == costonDefaultAttestors[0] {
+					rootHash := []byte("some_hash")
+					return rootHash, 0, nil
+				}
+
+				if caller.Address() != stateConnectorCoinbaseSignalAddr(params.CostonChainID, blockTime) {
+					return nil, 0, errors.New("caller address should be state connector coinbase signal addr")
+				}
+
+				if context.Coinbase == testCoinbaseAddress {
+					return nil, 0, errors.New("original coinbase address should have been changed to state connector coinbase signal addr")
+				}
+				return nil, 0, nil
+			}
+
+			c := newConnector(mockSCC, mockMsg)
+			err := c.finalizePreviousRound(params.CostonChainID, blockTime, currentRoundNumber)
+			require.NoError(t, err)
+			assert.Equal(t, testCoinbaseAddress, mockSCC.context.Coinbase, "coinbase address should be changed to the original address")
+		})
+		t.Run("handles count attestations error", func(t *testing.T) {
+			t.Parallel()
+
+			mockMsg, mockSCC := buildStateConnectorMock("coston chain", testCoinbaseAddress)
+
+			mockSCC.CallFunc = func(context vm.BlockContext, caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+				if caller.Address() == costonDefaultAttestors[0] {
+					return nil, 0, errors.New("attestation error")
+				}
+
+				if caller.Address() != stateConnectorCoinbaseSignalAddr(params.CostonChainID, blockTime) {
+					return nil, 0, errors.New("caller address should be state connector coinbase signal addr")
+				}
+
+				if context.Coinbase == testCoinbaseAddress {
+					return nil, 0, errors.New("original coinbase address should have been changed to state connector coinbase signal addr")
+				}
+				return nil, 0, nil
+			}
+
+			c := newConnector(mockSCC, mockMsg)
+			err := c.finalizePreviousRound(params.CostonChainID, blockTime, currentRoundNumber)
+			require.NoError(t, err)
```

Comment:
How is the error handled here? Seems ignored to me if we require no error return

Reply:
Round finalization should not be affected when one of the attestors returns an error, thus `require.NoError(t, err)`